### DISCUSSION TITLE
Epiphany, a notion clone

### DISCUSSION THREAD

**Discussion by shi-yan** [tauri-apps/tauri#6498] [Show and tell]
Hello everyone,

I want to show you Epiphany, a notion-inspired editor.

Screen recording: ( [https://youtu.be/_C0rpXXaETc](https://youtu.be/_C0rpXXaETc)   )  

Github ([https://github.com/shi-yan/Epiphany](https://github.com/shi-yan/Epiphany))

<img width="1582" alt="Screenshot 2023-03-19 at 10 33 29 AM" src="https://user-images.githubusercontent.com/326807/226196690-b31e28b9-4043-4695-998d-15961101d66d.png">

This is my impulse build. I wanted to start a blog. Having used so many blogging platforms, I'm tired of migrating and losing content. I want to own my data and version control them. Although there are static site generators, I don't think working on raw markdown files is a good writing experience. I, therefore, wanted to have a nice editor that has a focus on technical writing (good support for equations and references) and saves plain text files (I'm using the [djot](https://djot.net/) format) with a built-in static site generator.

The workload of building this has grown way more than I estimated. I'm now exhausted. The current release contains a ton of bugs and is only barely usable. But I want to get the first version out. I have created so many abandoned projects, I hope this won't become one of them. I will force myself to keep using it and improving it.

Some future directions to pursue include

1. static site generator (not done in the current version)
2. AI assistant using ChatGPT
3. built-in version control and collaboration

I also want to support splittable windows/tabs (similar to chrome and sublime text), but I don't know if that's feasible using Tauri. 
Dragging and dropping files also seem to be a little limited in Tauri. It sends an event, but it doesn't give the dropping location. If say I have multiple drop areas, I will have to detect which one should receive the drop event.

**Comment by FabianLars** [comment]
That looks so cool! 🥳 
Got confused at first because `epiphany` is also the name of the webkitgtk browser we use to double check bugs (Gnome web) but anywayyy 😂 

If you don't mind opening a PR for it, we'd much appreciate such a project in https://github.com/tauri-apps/awesome-tauri/ !

**Comment by AugustSaintFreytag** [comment]
Out of curiosity, did you implement contextual menus yet? I didn’t find a solution when I was last interested in doing that, I imagine script-side control over opening and populating pop-up context menus that are displayed by the OS, not just hacked together to look like a menu. In other terms, given you’re on macOS, I expect a function like `async openMenu(items, position, …)` to create an `NSMenu` on the Tauri/Rust side and show it at a given position relative to the window or screen, including callbacks for interactions.

**Reply by shi-yan** [reply]
no, I didn't implement a native context menu. I have a menu implemented in js.

**Comment by Kamek437** [comment]
Looks great! Notion needs some competition right now.
